pyparsing/core.py

```diff
@@ -5594,17 +5591,15 @@ def copy(self) -> ParserElement:
     def _setResultsName(self, name, list_all_matches=False):
         if (
             __diag__.warn_name_set_on_empty_Forward
-            and Diagnostics.warn_name_set_on_empty_Forward
```

Comment:
Accepted - will probably need to wrap in `#nofmt/#fmt` markers to suppress black reformatting.